elasticapm/contrib/starlette/__init__.py

```diff
@@ -192,3 +194,15 @@ async def _request_finished(self, response: Response):
 
         result = "HTTP {}xx".format(response.status_code // 100)
         elasticapm.set_transaction_result(result, override=False)
+
+    def get_route_name(self, request: Request) -> str:
+        path = None
+        routes = request.scope["app"].routes
+        for route in routes:
+            match, _ = route.matches(request.scope)
+            if match == Match.FULL:
+                path = route.path
+                break
+            elif match == Match.PARTIAL and path is None:
+                path = route.path
+        return path
```

Comment:
Based on @HenrikOssipoff's comment, I wonder if `path` is `None` when we go to return, should we try appending a `"/"` to `path` to attempt to get a match that way? That *should* solve the matching issues he's seeing, right? Slightly more overhead, but I still think it's worth it to avoid all of those redirect one-off transaction names.